mock/api.js

```diff
@@ -1,27 +1,28 @@
 import moment from 'moment';
 import casual from 'casual';
+import createUniqueKey from '../src/utils/uniqueKey_generator';
 
 const DEFAULT_SIZE = 100;
 
 export const mockIndices = new Array(1).fill().map(() => moment().format('YYYY-MM'));
 
 export const mockControllers = new Array(DEFAULT_SIZE).fill().map((value, index) => ({
-  controller: `${casual.word}.${casual.word}.com`,
+  controller: `${createUniqueKey()}.com`,
   key: casual.ip,
   last_modified_string: moment().valueOf(moment.utc()),
   last_modified_value: moment.utc() + index,
   results: casual.integer(1, DEFAULT_SIZE),
 }));
 
 export const mockResults = hostname =>
-  new Array(DEFAULT_SIZE).fill().map(() => ({
+  new Array(DEFAULT_SIZE).fill().map((value, index) => ({
     '@metadata.controller_dir': hostname,
     config: casual.word,
     controller: hostname,
     end: moment.utc(),
     // Since dataset id is a long hex string, removed "-" characters here to make it look like real data
     id: casual.uuid.replace(/-/g, ''),
-    result: casual.word,
+    result: `${index}${hostname.slice(0, -6)}${index}`,
```

Comment:
This makes the data a bit more consistent; however `mockDetail` is still randomly generating the dataset name even though it's reflecting the requested controller/host name now. It would be good to fix that, as well.

I'm not really happy with the overall structure of the dashboard API mocks we've managed to accumulate, and it seems that each small step forward in stability and consistency has been a big step "sideways" in that we're really not reflecting the expected schema of the actual data in many ways. The dashboard tests are currently mostly insensitive to this, but I worry it's going to be an increasingly serious problem as the dashboard gets smarter.

Still, we're agile, and a small step forward remains a small step forward. With some residual reluctance, I'll approve this and just hope we can take more time to improve this further in the future.